controller/controller.go

```diff
@@ -226,6 +229,39 @@ func ListenAndServeEx(host string, port int, dir string, ln *net.Listener, http
 	return server.ListenAndServe(host, port, protected, handler, opened, closed, ln, http)
 }
 
+func (c *Controller) watchGC() {
+	var elapsed uint64
+	t := time.NewTicker(time.Second)
+	defer t.Stop()
+
+	for range t.C {
+		elapsed++
+		c.mu.RLock()
+		autoGC := c.config.AutoGC
+		c.mu.RUnlock()
+
+		func() {
+			if autoGC == 0 {
+				return
+			}
+
+			if c.stopWatchingAutoGC {
+				return
```

Comment:
This only returns from the inner function, but it should conclude the `watchGC()`.
Also is should be between `c.mu.RLock/RUnlock`.